daml/ContingentClaims/Financial.daml

```diff
@@ -49,11 +49,11 @@ fixed principal coupon = floating (O.pure principal) (O.pure coupon)
 european : t -> Claim t x a -> Claim t x a
 european maturity payoff = When (at maturity) (payoff `or` zero)
 
--- | Bermudan option on the passed claim.
-bermudan : Claim t x a -> [t] -> Claim t x a
-bermudan _ [] = zero
-bermudan c [t] = european t c
-bermudan c (t :: t' :: ts) = Or (european t c) (european t' c) (fmap (`european` c) ts) -- This is written like this for efficiency
+-- | Bermudan option on the passed claim. Given a pre-defined set of times {t_1, t_2, .., t_N}, it allows the holder to acquire the underlying claim on at most one of these times.
+bermudan : [t] -> Claim t x a -> Claim t x a
+bermudan [] _ = zero
+bermudan [t] c = european t c
+bermudan (t :: ts) c = When (at t) $ Or c (bermudan ts c) []
```

Comment:
```suggestion
bermudan (t :: ts) c = when (at t) (c `or` bermudan ts c)
```

I really meant to not export the upper-case constructors, but they're required for tests, and also if you want to pattern match in a fold. But maybe we should reconsider this.